scRNA-seq-advanced/04-differential_expression.Rmd

```diff
@@ -538,73 +595,66 @@ EnhancedVolcano::EnhancedVolcano(deseq_results,
   theme(legend.position = "bottom")
 ```
 
-Let's make some UMAPs where we plot a gene that's identified to be differentially expressed in a given cell type and see what the expression of that gene is across samples. 
-First we need to extract the expression for each of the genes we would like to plot from the `logcounts` matrix and combine with the `colData`.
 
-```{r}
-# add column to colData with expression of gene of interest 
-# here we will actually put all of the significant genes so then we can plot any of them
-sig_expressed_genes <- deseq_results %>%
-  dplyr::filter(pvalue < 1e-5,
-                abs(log2FoldChange) > 2) %>%
-  dplyr::pull(ensembl_id)
-
-sig_genes_counts <- logcounts(rms_sce[sig_expressed_genes,]) %>%
-  as.matrix() %>%
-  t() %>%
-  as.data.frame() %>%
-  dplyr::mutate(cell_id = colnames(rms_sce))
-  
-combined_coldata_df <- as.data.frame(colData(rms_sce)) %>%
-  dplyr::left_join(sig_genes_counts, by = "cell_id")
-
-# add back modified coldata containing gene expression for sig genes 
-colData(rms_sce) <- DataFrame(combined_coldata_df, row.names = combined_coldata_df$cell_id)
-```
-
-We will pick a gene and look at the expression of that gene across subdiagnosis, sample, and cell type.
+We can also return back to the SCE object that we used to create our pseudo-bulked SCE and look at gene expression of some of the significant genes. 
+We can create UMAP plots as we did previously but instead of labeling each cell with metadata we can color each of the cells by the level of gene expression of a specified gene. 
+We will also use some of the `ggplot2` skills we picked up earlier, like `facet_grid()` to plot cells from different RMS subtypes separately.
+This can help us validate the `DESeq2` results so that we can visualize gene expression changes across our cell type of interest on a single-cell level. 
 
 ```{r}
 # filter to just myoblast cells and remove any NA's before plotting
 myoblast_combined_sce <- rms_sce[, which(rms_sce$celltype_broad == "Tumor_Myoblast")]
 
-# first look at just ARMS vs. ERMS 
+# plot PTPRT expression in ARMS vs. ERMS
 scater::plotReducedDim(myoblast_combined_sce,
                        dimred = "fastmnn_UMAP",
                        colour_by = "ENSG00000196090", #PTPRT
                        point_size= 0.5,
                        point_alpha = 0.4,
                        other_fields = "diagnosis_group") +
-  facet_wrap(~ diagnosis_group, nrow = 3)
+  facet_grid(cols = vars(diagnosis_group))
 ```
 
+In the above plot we only plotted the tumor myoblast cells that we used in our DE analysis. 
+However, we might be interested to see the expression of genes that are differentially expressed in tumor myoblasts in other cell types present in our samples.
 
 ```{r}
-# let's look across some other cell types
+# let's compare gene expression across some other cell types
```

Comment:
any reason why these 4? Well, tumor is implicit, but why this specific normal tissue?